src/custom/pages/Claim/types.ts

```diff
@@ -7,3 +8,11 @@ export type ClaimCommonTypes = {
   tokenCurrencyAmount: CurrencyAmount<Token>
   handleChangeAccount: (e: SyntheticEvent<HTMLButtonElement>) => void
 }
+
+export type ParsedUserClaim = UserClaimData & {
+  parsedAmount: string
+  isFree: boolean
+  currency: string
+  price: number
```

Comment:
I don't think the price will be a number. Numbers have floating point errors. 
We should have it as weis

We should have it exposed in the units of the token itself.  Maybe a `CurrencyAmount<Token>`?

Reply:
Yep, we shouldn't be using `number` for any amounts.
Safest bet is to work with `string`, or if you want to be fancy the `CurrencyAmount` or similar classes as Anxo mentions

Reply:
Changed to CurrencyAmount<Token> | undefined